File: hse/BPM161/alferov/2/src/test/java/net/vasalf/hw/lockfree/LinCheckTest.java
package net.vasalf.hw.lockfree;

import com.devexperts.dxlab.lincheck.LinChecker;
import com.devexperts.dxlab.lincheck.annotations.Operation;
import com.devexperts.dxlab.lincheck.annotations.Param;
import com.devexperts.dxlab.lincheck.paramgen.IntGen;
import com.devexperts.dxlab.lincheck.strategy.stress.StressCTest;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

@StressCTest
public class LinCheckTest {
    private MySet<Integer> set = new LockFreeSet<>();

    @Operation
    public boolean add(@Param(gen = IntGen.class) int item) {
        return set.add(item);
    }

    @Operation
    public boolean remove(@Param(gen = IntGen.class) int item) {
        return set.remove(item);
    }

    @Operation
    public boolean isEmpty() {
        return set.isEmpty();
    }

    @Operation
    public boolean contains(@Param(gen = IntGen.class) int item) {
        return set.contains(item);
    }

    @Operation
    public List<Integer> toList() {
        List<Integer> ret = new ArrayList<>();
        Iterator<Integer> iterator = set.iterator();
        while (iterator.hasNext()) {
            ret.add(iterator.next());
        }
        return ret;
    }

    @Test
    public void test() {
        LinChecker.check(LinCheckTest.class);
    }
}

File: hse/BPM161/kuporosov/2/src/test/kotlin/net/netau/vasyoid/LincheckTest.kt
package net.netau.vasyoid

import com.devexperts.dxlab.lincheck.LinChecker
import com.devexperts.dxlab.lincheck.annotations.Operation
import com.devexperts.dxlab.lincheck.strategy.stress.StressCTest
import org.junit.Test

@StressCTest
class LincheckTest {
    private val set = LockFreeSet<Int>()

    @Operation
    fun add(value: Int) = set.add(value)

    @Operation
    fun contains(value: Int) = set.contains(value)

    @Operation
    fun isEmpty() = set.isEmpty()

    @Operation
    fun remove(value: Int) = set.remove(value)

    @Operation
    fun iterator(): List<Int> = set.iterator().asSequence().toList()

    @Test
    fun test() {
        LinChecker.check(LincheckTest::class.java)
    }
}
